lib/three_scale/domain_substitution.rb

```diff
@@ -151,7 +151,6 @@ def match_internal_domain?(host)
     end
 
     deprecate domain: "use #internal_domain or #external_domain",
-      self_domain: "use #internal_admin_domain or #external_admin_domain",
```

Comment:
I think this should stay unless we remove the method or make it private (the latter seems to make more sense given this is an attribute actually).